flybywiresim/a32nx
# feat(lights): add ANN LT DIM functionality

## Summary of Changes
Adds DIM functionality to OVHD ANN LT. 

Logic for TEST is refactored and uses 1 variable: `A32NX_OVHD_INTLT_ANN` for TEST/BRT instead of 2. This was needed in order to make a nice implementation of DIM.

Variables `XMLVAR_SWITCH_OVHD_INTLT_ANNLT_Position` and `XMLVAR_LTS_Test` is removed.

When OVHD ANN LT switch is set to DIM, every korry switch, led, green bar button and annunciator light should dim in intensity.



## Screenshots



**UPPER OVHD/OVHD/COCKPIT BRT/DIM**
![UPPER-OVHD-DIM-BRT](https://user-images.githubusercontent.com/433429/119724902-9a0b2100-be6f-11eb-983c-817c65205503.gif)
![OVHD-DIM-BRT](https://user-images.githubusercontent.com/433429/119724812-7ba52580-be6f-11eb-8b87-7d1ecb686b89.gif)
![COCKPIT-DIM-BRT](https://user-images.githubusercontent.com/433429/119724959-aa230080-be6f-11eb-8a1a-32627f811475.gif)


## References


> In one sentence: every light that comes on during the light test is also influenced by dim/brt position, including the green bars of FCU and RMP, the green LEDs and the amber SEL indicator and, to my own surprise, even the white lights of the reception knobs of the ACP. The latter is probably depending on the installed part number. There are a few different versions out there. None of them matches our model 100%, but for the one that looks the most similar to ours and is also the newest variant, this is how it behaves. (@komp1821)

https://user-images.githubusercontent.com/433429/119434188-f6f3c380-bd17-11eb-8d21-190f310a1bd0.mp4

> Every lit-up Pushbutton switch 'indication' gets dimmed, except those bloody radio knob volume lights. FCU knobs, radio buttons etc have their own dimmers. PB switches also have their panel's backlighting dimmers (separate from the 'indication' it shows). From memory, if you select DIM, it will dim the switches indications: eg: Fuel Pump switches "OFF" would dim. Landing gear lights dim. APU "AVAIL" and "START" indications would dim. The yellow/orange backlighting however is controlled by the rotary panel dimmer switch. (@Parorng[Z+7])

https://user-images.githubusercontent.com/433429/119434212-0115c200-bd18-11eb-88d7-b484565e9e2f.mov


Discord username (if different from GitHub): bouveng&#35;1767

## Testing instructions

Note: 

Testing instructions is just a “serving suggestion” and a way to light as many buttons as possible. One can test this how one seem fit and meaningful. Many lights, buttons and annunciators share logic so not needed to test every instance.

One can also reference the animated gifs above to get an idea.

1. Spawn anywhere, cold and dark. Preferably a "dark" airport.
2. Set time to something dark/sunset. Dark enough to see lights, light enough to have some sort of ambient light in the cockpit.

OVERHEAD SETUP:

3. Set EXT PWR to ON.
4. CALLS EMER to ON,
5. GND CTL to ON. 
4. GPWS, TERR, SYS, G/S, FLAP, LDG to OFF.
5. GEN1 LINE to OFF.
6. FIRE ENG1+2 PUSH, ENG 1+2AGENT discharge.
7. FIRE APU PUSH, discharge.
9. ENG1 + 2 FUEL+HYD PUMPS, all off. XFEED on.
10. COMMERCIAL + GALY + CAB to OFF.
11. GEN1 to OFF.
12. APU GEN to OFF.
11. PACK1+2 to OFF.
12. ENG1+2 BLEED to OFF.
13. HOT AIR to OFF.
14. APU BLEED to ON.
15. WING, ENG1+2, PROBE/WIN/HEAT ANTIICE to ON.
16. CABIN PRESS MODE SEL to MAN.
17. APU MASTER SW to ON,
18. EMER EXIT LT to OFF.
19. ENG MAN START, 1+2 to ON.
20. VENTILATION BLOWER, EXTRACT, and CAB FANS to OFF.
21. UPPER OVHD APU AUTO EXITING TEST to ON.
22. UPPER OVHD ENG1+2 FADEC GND PWR to ON.
23. UPPER OVHD BLUE PUMP OVRD to OFF.
24. HYD LEAK MEASUREMENT VALVES, G+B+Y to OFF.
25. OXYGEN TMR RESET to ON.
26. SVGE INT OVRD to ON.
27. AVIONICS COM LT to ON.

COCKPIT SETUP:

28. FD 1+2 and LS 1+2 to ON.
29. CSTR 1+2 to ON.
30. TERR ON ND 1 + 2 to ON.
31. BRK FAN to ON.
32. AUTO BRK to MAX.
33. ECAM PAGE, select ELEC.
34. RADIO select VHF2.
35. RADIO select VHFC transmitter.
36. RADIO L+R toggle VHF3 audio to ON.
37. Throttle to full.

Then proceed to look around in the cockpit, alternating between DIM and BRT.

Overhead, Upper overhead, green bar buttons on autopilot, ecam buttons, leds, buttons and volume knobs on radios, cockpit door annunicator, master warning/caution, terr on nd, gear, autobrake and brake fan annunciators.s

Bonus test: If one want to verify the MASTER WARNING. Spawn on RWY. full throttle with parking brakes set.
Bonus test: If one want to verify the AP1+AP2, EXPED, LOC and APPR, take off and set up for an approach.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
